packages/pg/lib/client.js

```diff
@@ -641,14 +712,21 @@ class Client extends EventEmitter {
       this._connect((error) => {
         if (error) {
           logger.silly("Not able to connect to " + this.host + " due to error " + error.message)
-          if (this.connectionParameters.loadBalance && Client.hostServerInfo.size !== 0) {
-            if (Client.hostServerInfo.has(this.host)) {
+          if (this.connectionParameters.loadBalance !== 'false' && (Client.hostServerInfoPrimary.size !== 0 || Client.hostServerInfoRR.size !== 0)) {
+            if (Client.hostServerInfoPrimary.has(this.host)) {
+              logger.debug("Adding " + this.host + " to failed host list")
+              Client.failedHosts.set(this.host, Client.hostServerInfoPrimary.get(this.host))
+              let start = new Date().getTime();
+              Client.failedHostsTime.set(this.host, start)
+              Client.connectionMap.delete(this.host)
+              Client.hostServerInfoPrimary.delete(this.host)
+            } else if (Client.hostServerInfoRR.has(this.host)) {
               logger.debug("Adding " + this.host + " to failed host list")
-              Client.failedHosts.set(this.host, Client.hostServerInfo.get(this.host))
+              Client.failedHosts.set(this.host, Client.hostServerInfoRR.get(this.host))
               let start = new Date().getTime();
               Client.failedHostsTime.set(this.host, start)
               Client.connectionMap.delete(this.host)
-              Client.hostServerInfo.delete(this.host)
+              Client.hostServerInfoRR.delete(this.host)
             } else if (Client.failedHosts.has(this.host)) {
               logger.silly("Removing host " + this.host + " from failed host list")
```

Comment:
Why do we remove the host from failedHostList here? Please add a note here as a todo if we don't the answer.

Reply:
Note added.